yrs/src/block_store.rs

```diff
@@ -125,54 +135,68 @@ impl BlockStore {
-                    } else {
-                        types::TypePtr::Id(block::BlockPtr::from(update_decoder.read_left_id()))
-                    };
-                    let parent_sub = if cantCopyParentInfo && info & 0b00100000 > 0 {
-                        Some(update_decoder.read_string())
-                    } else {
-                        None
-                    };
-                    let item: block::Item = todo!();
-                    structs.list.push(block::Block::Item(item));
-                    clock += item.len();
-
-                } else {
-                    // is a GC
-                    let len: u32 = update_decoder.rest_decoder.read_var_uint();
-                    let skip = block::GC {
-                        id,
-                        len
-                    };
-                    structs.list.push(block::Block::GC(skip));
-                    clock += len;
+                match info {
+                    BLOCK_SKIP_REF_NUMBER => {
+                        let len: u32 = update_decoder.rest_decoder.read_var_uint();
+                        let skip = block::Skip { id, len };
+                        blocks.list.push(block::Block::Skip(skip));
+                        clock += len;
+                    }
+                    BLOCK_GC_REF_NUMBER => {
+                        let len: u32 = update_decoder.rest_decoder.read_var_uint();
+                        let skip = block::GC { id, len };
+                        blocks.list.push(block::Block::GC(skip));
+                        clock += len;
+                    }
+                    info => {
+                        let cant_copy_parent_info = info & (HAS_ORIGIN | HAS_RIGHT_ORIGIN) == 0;
+                        let origin = if info & HAS_ORIGIN != 0 {
+                            Some(update_decoder.read_left_id())
+                        } else {
+                            None
+                        };
+                        let right_origin = if info & HAS_RIGHT_ORIGIN != 0 {
+                            Some(update_decoder.read_right_id())
+                        } else {
+                            None
+                        };
+                        let parent = if cant_copy_parent_info {
+                            types::TypePtr::Named(update_decoder.read_string().to_owned())
+                        } else {
+                            types::TypePtr::Id(block::BlockPtr::from(update_decoder.read_left_id()))
+                        };
+                        let parent_sub = if cant_copy_parent_info && (info & 0b00100000 != 0) {
+                            Some(update_decoder.read_string().to_owned())
+                        } else {
+                            None
+                        };
+                        let content =
+                            ItemContent::decode(update_decoder, info, BlockPtr::from(id.clone())); //TODO: What BlockPtr here is supposed to mean
```

Comment:
This is a reference to the `item` itself. If this is a type, we need some kind of reference to the type. A `TypePtr` will point to an `item` that contains a type.